File: ecommerce-angular/src/app/module/cart/item.ts
import { Product } from "../product/product";


export class Item implements Omit<Product, 'size' |'color'>   {
    id: number;
    img: string;
    brand: string;
    title: string;
    selectedColor: string;
    price: number;
    selectedSize: string;
    quantity: number;

    constructor(p:Product, size: string, color: string){
        this.id=p.id
        this.img = p.img
        this.brand = p.brand
        this.title = p.title
        this.selectedColor = color
        this.price = p.price
        this.selectedSize = size
        this.quantity = 1
    }

}

File: ecommerce-spring/ecommerce/src/main/java/com/ecommerce/model/Item.java
package com.ecommerce.model;
import com.ecommerce.utils.Color;
import com.ecommerce.utils.Size;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "ITEMS")
public class Item {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "img")
    private String img;

    @Column(name = "title")
    private String title;

    @Column(name = "brand")
    private String brand;

    @Column(name = "price")
    private Float price;

    @Column(name = "size")
    private Size[] sizes;

    @Column(name = "color")
    private Color[] colors;

    public Long getId(){
        return this.id;
    }

    public String getImg(){
        return this.img;
    }

    public Size[] getSizes(){
        return this.sizes;
    }

    public Color[] getColors(){
        return this.colors;
    }

}
